test/parallel/test-tls-sni-server-client.js

```diff
@@ -20,17 +20,17 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
-const common = require('../common');
-if (!common.hasCrypto)
-  common.skip('missing crypto');
+const { hasCrypto, mustCall, mustNotCall, skip } = require('../common');
+if (!hasCrypto)
+  skip('missing crypto');
 
-const assert = require('assert');
-const tls = require('tls');
-const fixtures = require('../common/fixtures');
+const { strictEqual } = require('assert');
+const { connect, createServer } = require('tls');
+const { readKey } = require('../common/fixtures');
```

Comment:
Not blocking, but why the mass-introduction of destructuring?  To me, it makes the subsequent code harder to understand. Later on, if I see `fixtures.readKey()`, I know exactly what that means. But if I just see `readKey()`...I'm less sure. It's even words for `connect()` on the line above.

Again, not blocking because it's ultimately a style choice, but I'm genuinely curious if there's a motivation/advantage I'm not seeing that I should be seeing.

Reply:
No particular reason, in most cases it obvious and shorter. For example `assert.strictEqual` why not only `strictEqual` and the same for `createServer`, `mustCall`, `mustNotCall`. The others are for consistency.

I will revert this to make the diff smaller.

Reply:
When I was working in python (which js's import syntax seems to be modelled on), we forbade this usage in our coding standards. Functions were always called with their leading module name, `tls.connect` not `connect` (which could be `net.connect`). We found it makes code harder to understand when we couldn't see where a function came from. It seems to me that the js community, or perhaps node, is going the opposite way. Its a stylistic choice, I don't feel strongly enough about it to try to change it, but I'm in agreement with @Trott that eliding the module name makes subsequent code harder to read.

Reply:
I partially agree but it in my opinion it depends. On a large file it makes sense. When there are just few lines of code, it's easy to see where it's coming from.

One is sure though, I don't like to mix things, so it's always or never for me :)